File: Examples/test-suite/perl5/cpp_static_runme.pl
use strict;
use warnings;
use Test::More tests => 6;
BEGIN { use_ok('cpp_static') }
require_ok('cpp_static');

can_ok('cpp_static::StaticFunctionTest', 'static_func');
cpp_static::StaticFunctionTest->static_func();
can_ok('cpp_static::StaticFunctionTest', 'static_func_2');
cpp_static::StaticFunctionTest->static_func_2(1);
can_ok('cpp_static::StaticFunctionTest', 'static_func_3');
cpp_static::StaticFunctionTest->static_func_3(1, 2);
$cpp_static::StaticMemberTest::static_int = 10;
is($cpp_static::StaticMemberTest::static_int, 10);

File: Examples/test-suite/ruby/cpp_static_runme.rb
#!/usr/bin/env ruby
#
# This test implementation is derived from its Python counterpart.
#

require 'swig_assert'
require 'cpp_static'

Cpp_static::StaticFunctionTest.static_func()
Cpp_static::StaticFunctionTest.static_func_2(1)
Cpp_static::StaticFunctionTest.static_func_3(1, 2)

swig_assert_equal("Cpp_static::StaticMemberTest.static_int", "99", binding)
Cpp_static::StaticMemberTest.static_int = 10
swig_assert_equal("Cpp_static::StaticMemberTest.static_int", "10", binding)

swig_assert_equal("Cpp_static::StaticBase.statty", "11", binding)
swig_assert_equal("Cpp_static::StaticBase.grab_statty_base", "11", binding)
swig_assert_equal("Cpp_static::StaticDerived.statty", "111", binding)
swig_assert_equal("Cpp_static::StaticDerived.grab_statty_derived", "111", binding)
Cpp_static::StaticBase.statty = 22
Cpp_static::StaticDerived.statty = 222
swig_assert_equal("Cpp_static::StaticBase.statty", "22", binding)
swig_assert_equal("Cpp_static::StaticBase.grab_statty_base", "22", binding)
swig_assert_equal("Cpp_static::StaticDerived.statty", "222", binding)
swig_assert_equal("Cpp_static::StaticDerived.grab_statty_derived", "222", binding)
